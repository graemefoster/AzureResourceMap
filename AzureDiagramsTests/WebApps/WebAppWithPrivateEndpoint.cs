using AzureDiagrams.Resources;
using AzureDiagrams.Resources.Retrievers.Extensions;
using Shouldly;

namespace AzureDiagramsTests.WebApps;

public class WebAppWithPrivateEndpoint
{
    [Fact]
    public async Task CanDrawDiagram()
    {
        var resources = BuildScenario();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources,
            false,
            false,
            false,
            true,
            false);

        diagram.ShouldMatchApproved();
    }

    
    [Fact]
    public async Task CanDrawCondensedDiagram()
    {
        var resources = BuildScenario();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources,
            true,
            false,
            false,
            true,
            false);

        diagram.ShouldMatchApproved();
    }

    private static AzureResource[] BuildScenario()
    {
        var vnetId = "/subscriptions/123/resourceGroups/rg1/providers/microsoft.network/virtualnetworks/vnet123";

        var vnet = new VNet(vnetId, "test-vnet", new[]
        {
            new VNet.Subnet("private-endpoints", "10.1.1.0/24"),
            new VNet.Subnet("vnet-integration", "10.1.2.0/24"),
        });

        var appServicePlanId =
            "/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/serverFarms/TestAppServicePlan";

        var asp = new AppServicePlan(appServicePlanId, "TestAppServicePlan");

        var peId = new Guid("36a566f9-f6b8-439c-9371-204842160e2b").ToString();
        var pe2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2d").ToString();

        var app1HostName = "TestApp1.azurewebsites.net";
        var app2HostName = "TestApp2.azurewebsites.net";

        var app1 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp1",
            appServicePlanId,
            "TestApp1",
            false,
            Array.Empty<string>(),
            new[] { app1HostName },
            virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
            new PrivateEndpointExtensions(new[] { peId }));

        var app2 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp2",
            appServicePlanId,
            "TestApp2",
            false,
            new[] { $"https://{app1HostName}" },
            new[] { app2HostName },
            virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
            new PrivateEndpointExtensions(new[] { pe2Id }));

        var nicId = new Guid("36a566f9-f6b8-439c-9371-204842160e2c").ToString();
        var nic = new Nic(nicId,
            IpConfigurations.ForPrivateEndpoint("10.1.1.1", $"{vnetId}/subnets/private-endpoints", app1HostName));

        var privateEndpoint = new PrivateEndpoint(peId, new[] { $"{vnetId}/subnets/private-endpoints" },
            new[] { nic.Id }, new[] { $"https://{app1HostName}" });

        var nic2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2e").ToString();
        var nic2 = new Nic(nic2Id,
            IpConfigurations.ForPrivateEndpoint("10.1.1.2", $"{vnetId}/subnets/private-endpoints", app2HostName));

        var privateEndpoint2 = new PrivateEndpoint(pe2Id, new[] { $"{vnetId}/subnets/private-endpoints" },
            new[] { nic2.Id }, new[] { $"https://{app2HostName}" });

        var azureResources = new AzureResource[]
        {
            vnet,
            asp,
            app1,
            app2,
            nic,
            nic2,
            privateEndpoint,
            privateEndpoint2
        };

        var resources = azureResources.Process();
        return resources;
    }
}