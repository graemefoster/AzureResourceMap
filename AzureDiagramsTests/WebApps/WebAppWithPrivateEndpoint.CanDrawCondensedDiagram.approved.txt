<mxGraphModel>
	<root>
		<mxCell id="0" />
		<mxCell id="1" parent="0" />
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration.empty" value="&lt;br/&gt;(Subnet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#ffffff;verticalAlign=middle" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration">
    <mxGeometry x="50" y="50" width="150" height="75" 
    as="geometry" />
</mxCell>
<mxCell id="66f228a5-2df7-c23b-4814-95adb050732d" value="TestApp1" style="html=1;image;image=img/lib/azure2/app_services/App_Services.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="50" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="f1921020-766f-e0b6-735f-33e982981b99" value="TestApp2" style="html=1;image;image=img/lib/azure2/app_services/App_Services.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="50" y="125" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints" value="private-endpoints&lt;br/&gt;(10.1.1.0/24)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="50" y="50" width="150" height="235" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration" value="vnet-integration&lt;br/&gt;(10.1.2.0/24)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="300" y="50" width="250" height="185" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c" value="test-vnet&lt;br/&gt;(VNet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F8CECC;verticalAlign=top" vertex="1" parent="207c67fc-34bc-d9f0-369e-0f16975fc114.container">
    <mxGeometry x="50" y="50" width="600" height="345" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.image.0" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="560" y="305" width="30" height="30" 
    as="geometry" />
</mxCell>
<mxCell id="207c67fc-34bc-d9f0-369e-0f16975fc114.container" value="global&lt;br/&gt;(Region)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F5F5F5;verticalAlign=top" vertex="1" parent="1">
    <mxGeometry x="0" y="0" width="700" height="455" 
    as="geometry" />
</mxCell>
<mxCell id="32917fb3-76cd-325d-0425-483011b23905" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="f1921020-766f-e0b6-735f-33e982981b99" target="66f228a5-2df7-c23b-4814-95adb050732d">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             <mxCell id="32917fb3-76cd-325d-0425-483011b23905--1" value="calls" style="edgeLabel;html=1;align=center;verticalAlign=middle;resizable=0;points=[];" vertex="1" connectable="0" parent="32917fb3-76cd-325d-0425-483011b23905">
             <mxGeometry x="-0.5" relative="1" as="geometry">
             <mxPoint as="offset" />
             </mxGeometry>
             </mxCell>

	</root>
</mxGraphModel>